tests/test_warm_reboot.py

```diff
@@ -780,3 +789,687 @@ def test_swss_port_state_syncup(dvs, testlog):
+    dvs.runcmd("sysctl net.ipv6.conf.all.disable_ipv6=0")
+
+    dvs.runcmd("ifconfig {} 111.0.0.1/24 up".format(intfs[0]))
+    dvs.runcmd("ip -6 addr add 1110::1/64 dev {}".format(intfs[0]))
+
+    dvs.runcmd("ifconfig {} 122.0.0.1/24 up".format(intfs[1]))
+    dvs.runcmd("ip -6 addr add 1220::1/64 dev {}".format(intfs[1]))
+
+    dvs.runcmd("ifconfig {} 133.0.0.1/24 up".format(intfs[2]))
+    dvs.runcmd("ip -6 addr add 1330::1/64 dev {}".format(intfs[2]))
+
+    time.sleep(1)
+
+    #
+    # Setting peer's ip-addresses and associated neighbor-entries
+    #
+    ips = ["111.0.0.2", "122.0.0.2", "133.0.0.2"]
+    v6ips = ["1110::2", "1220::2", "1330::2"]
+    macs = ["00:00:00:00:11:02", "00:00:00:00:12:02", "00:00:00:00:13:02"]
+
+    for i in range(len(ips)):
+        dvs.runcmd("ip neigh add {} dev {} lladdr {}".format(ips[i], intfs[i%2], macs[i]))
+
+    for i in range(len(v6ips)):
+        dvs.runcmd("ip -6 neigh add {} dev {} lladdr {}".format(v6ips[i], intfs[i%2], macs[i]))
+
+    time.sleep(1)
+
+    #
+    # Defining baseline IPv4 non-ecmp route-entries
+    #
+    dvs.runcmd("ip route add 192.168.1.100/32 nexthop via 111.0.0.2")
+    dvs.runcmd("ip route add 192.168.1.200/32 nexthop via 122.0.0.2")
+    dvs.runcmd("ip route add 192.168.1.300/32 nexthop via 133.0.0.2")
+
+    #
+    # Defining baseline IPv4 ecmp route-entries
+    #
+    dvs.runcmd("ip route add 192.168.1.1/32 nexthop via 111.0.0.2 nexthop via 122.0.0.2 nexthop via 133.0.0.2")
+    dvs.runcmd("ip route add 192.168.1.2/32 nexthop via 111.0.0.2 nexthop via 122.0.0.2 nexthop via 133.0.0.2")
+    dvs.runcmd("ip route add 192.168.1.3/32 nexthop via 111.0.0.2 nexthop via 122.0.0.2")
+
+    #
+    # Defining baseline IPv6 non-ecmp route-entries
+    #
+    dvs.runcmd("ip -6 route add fc00:11:11::1/128 nexthop via 1110::2")
+    dvs.runcmd("ip -6 route add fc00:12:12::1/128 nexthop via 1220::2")
+    dvs.runcmd("ip -6 route add fc00:13:13::1/128 nexthop via 1330::2")
+
+    #
+    # Defining baseline IPv6 ecmp route-entries
+    #
+    dvs.runcmd("ip -6 route add fc00:1:1::1/128 nexthop via 1110::2 nexthop via 1220::2 nexthop via 1330::2")
+    dvs.runcmd("ip -6 route add fc00:2:2::1/128 nexthop via 1110::2 nexthop via 1220::2 nexthop via 1330::2")
+    dvs.runcmd("ip -6 route add fc00:3:3::1/128 nexthop via 1110::2 nexthop via 1220::2")
+
+    time.sleep(5)
+
+    dvs.runcmd("swssloglevel -l INFO -c fpmsyncd")
+
+
+    #############################################################################
+    #
+    # Testcase 1. Having routing-warm-reboot disabled, restart zebra and verify
+    #             that the traditional/cold-boot logic is followed.
+    #
+    #############################################################################
+
+    # Restart zebra
+    dvs.stop_zebra()
+    marker1 = dvs.add_log_marker("/var/log/swss/swss.rec")
+    marker2 = dvs.add_log_marker("/var/log/swss/sairedis.rec")
+    dvs.start_zebra()
+
+    time.sleep(5)
+
```

Comment:
You need to make sure the state is reconciled per your diffs.

Reply:
See that in this initial testcase, warm-reboot is not enabled yet, so there's no warm-reboot FSM state per-se, as those are only applicable to apps where warm-reboot has been activated.

Reply:
Shouldn't you then clear any leftover state in checkAndStart if warm-restart is not activated and check against that?

Reply:
done